pages/partials/instructorFileList.ejs

```diff
@@ -0,0 +1,76 @@
+<% if (files_to_list.length > 0) { %>
+<script>
+  $(function() {
+      let showListElement = $("#toggle-list-of-files-<%= file_label %>");
+      showListElement.click(() => {
+          showListElement.text(showListElement.text() == "Show <%= file_label.toLowerCase() %> files" ? "Hide <%= file_label.toLowerCase() %> files" : "Show <%= file_label.toLowerCase() %> files");
+      });
+  });
+</script>
+<% } %>
+
+<th>
+  <%= file_label %> Files
+  <a class="btn btn-xs" role="button" data-placement="auto" data-trigger="focus" data-toggle="popover" data-html="true" title="<%= file_label %> Files" tabindex="0" data-content="<%= file_info %>"><span><i class="far fa-question-circle" aria-hidden="true"></i></span></a>
+</th>
+<td>
+  <div class="small">
+    <div class="row justify-content-start">
+      <div class="col">
+      <% if ((authz_data.has_course_permission_edit) && (! course.options.isExampleCourse)) { %>
+      <button id="instructorFileUploadForm-New<%= file_label %>" tabindex="0" class="btn btn-xs btn-secondary" role="button" href="#" data-toggle="popover" data-container="body" data-html="true" data-placement="auto" title="Upload file" data-content="<%= include('instructorFileUploadForm', {file: {id: "New" + file_label}}) %>" data-trigger="manual" onclick="$(this).popover('show')">
+        <i class="fa fa-plus"></i>
+        <span>Add new <%= file_label.toLowerCase() %> file</span>
+      </button>
+      <% } %>
+      <% if (files_to_list.length > 0) { %>
+      <a class="btn btn-xs btn-secondary" data-toggle="collapse" href="#list-of-files-<%= file_label %>" role="button" id="toggle-list-of-files-<%= file_label %>">Hide <%= file_label.toLowerCase() %> files</a>
+      <% } %>
+      </div>
+    </div>
+    <% if (files_to_list.length > 0) { %>
+    <div class="row justify-content-start collapse show pt-2" id="list-of-files-<%= file_label %>">
+      <div class="col">
+        <ul class="list-group list-group-flush">
+          <% files_to_list.forEach(function(f) { %>
+          <li class="list-group-item px-0 py-1">
+            <span class="pr-2 text-monospace"><%= f.name %></span>
```

Comment:
You may well be already implementing this as part of your "file browser" code, but it would be awesome if we could view files by clicking on their filenames, like happens in the GitHub code browser. I'm imagining that this "file view page" would also have an "edit" button, etc, on it. I think this would be very helpful to allow viewing without the risk of editing.

Reply:
Done, done, done, etc.